api/src/test/java/org/pmiops/workbench/api/BigQueryBaseTest.java

```
@@ -0,0 +1,107 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+@SpringBootTest(classes = {TestBigQueryConfig.class})
+public abstract class BigQueryBaseTest {
+
+    private static final Logger log = Logger.getLogger(BigQueryBaseTest.class.getName());
+
+    @Autowired
+    BigQuery bigquery;
+
+    @Autowired
+    WorkbenchConfig workbenchConfig;
+
+    public static final String BASE_PATH = "src/test/resources/bigquery/";
+
+    @Before
+    public void setUp() throws Exception {
+        createDataSet(workbenchConfig.bigquery.dataSetId);
+        for (String tableName: getTableNames()) {
+            createTable(workbenchConfig.bigquery.dataSetId, tableName);
+            insertData(workbenchConfig.bigquery.dataSetId, tableName);
+        }
+    }
+
+    @After
+    public void tearDown() {
+        for (String tableName: getTableNames()) {
+            deleteTable(workbenchConfig.bigquery.dataSetId, tableName);
+        }
+        deleteDataSet(workbenchConfig.bigquery.dataSetId);
+    }
+
+    public abstract List<String> getTableNames();
+
+    private void createDataSet(String dataSetId) {
+        Dataset dataSet = bigquery.create(DatasetInfo.newBuilder(dataSetId).build());
+    }
+
+    private void createTable(String dataSetId, String tableId) throws Exception {
+        ObjectMapper jackson = new ObjectMapper();
+        String rawJson =
+                new String(Files.readAllBytes(Paths.get(BASE_PATH + "schema/" + tableId.toLowerCase() + ".json")), Charset.defaultCharset());
+        JsonNode newJson = jackson.readTree(rawJson);
+
+        Gson gson = new Gson();
+        Schema schema = gson.fromJson(newJson.toString(), Schema.class);
+        StandardTableDefinition tableDef = StandardTableDefinition.of(schema);
+        bigquery.create(TableInfo.of(TableId.of(dataSetId, tableId), tableDef));
+    }
+
+    private boolean insertData(String dataSetId, String tableId) throws Exception {
```

Rather than returning booleans from all of these, we should probably have them throw exceptions if anything fails. We don't want to have to check the return value everywhere I expect.

---

throwing exceptions now.